### DISCUSSION TITLE
Distortion saving a PNG as GIF

### DISCUSSION THREAD

**Discussion by anonymous-sushi** [python-pillow/Pillow#7030] [Q&A] [answered]
Hello!
I have a PNG image that I am overlaying a transparent gif over it.
But even without overlaying the gif step I am just testing out saving the PNG as a GIF and it is distorting the image already. How do i preserve the image while saving it as a GIF?

        background_image = Image.open("MY_IMAGE.png")
	animated_gif  = Image.open(GIF_FILE)

	frames = []

	final_frame_count = animated_gif.n_frames
	print("Gif Frames",final_frame_count)
	startTime = time.time()

        #This already results in a distorted image
	**background_image .save("test-1.gif", format='GIF', save_all=False, loop=0, disposal=2)**

	for i in range(final_frame_count):
		animated_gif.seek(i % final_frame_count)
		
		frame = background_image.copy()
		frame.paste(animated_gif, (810,520), mask=animated_gif.convert("RGBA"))				
		frames.append(frame)

**Comment by Yay295** [comment]
Can you share the images you're using?

**Reply by anonymous-sushi** [reply]
Not atm, but the image basically has a gradient and i think that exceeds the max 255 colors per gif frame? How would i preserve the colors if i save the png as GIF, or after i overlay a transparent gif on top of the png and save it as gif

**Reply by radarhere** [reply, marked as answer]
If have more than 256 new colors in a single frame, then you can't perfectly translate that to a GIF. That's just how the GIF format works.

https://en.wikipedia.org/wiki/GIF
> The format supports up to 8 bits per pixel for each image, allowing a single image to reference its own palette of up to 256 different colors

I say 'new colors' because you could have 256 colors in the first frame, and then 256 colors added in the second frame, leading to the second image in the sequence having 512 colors - but that would involve carefully crafting the image, which I doubt is what you want.

**Reply by anonymous-sushi** [reply]
Is there a way to convert a PNG to gif friendly colors through pillow without distorting the image too much?

**Reply by radarhere** [reply]
Here's some code that generates a image with more than 256 characters, by creating two simple gradients, and then saves it as a PNG and a GIF.
```python
from PIL import Image
im = Image.new("RGB", (300, 300))
for x in range(300):
	for y in range(300):
		c = (x, 0, 0) if x < 150 else (0, 150 - (x - 150), 0)
		im.putpixel((x, y), c)
im.save("gradient.png")
im.save("gradient.gif")
```
![gradient](https://user-images.githubusercontent.com/3112309/226219301-ef3f8634-376d-420f-8090-92e4b46a55c3.png)
![gradient](https://user-images.githubusercontent.com/3112309/226219216-cf04e6f0-3fde-4d61-b951-4a575aa65d5c.gif)

Let us know if you think otherwise, but comparing the two images, that doesn't look very distorted to me.

So without a specific image, I'm not sure how to answer your question.
